cpp/api/include/trtorch/trtorch.h

```diff
@@ -404,4 +433,16 @@ TRTORCH_API torch::jit::Module CompileGraph(const torch::jit::Module& module, Ex
  * @return: std::string: Serialized TensorRT engine equivilant to the method graph
  */
 TRTORCH_API std::string ConvertGraphToTRTEngine(const torch::jit::Module& module, std::string method_name, ExtraInfo info);
+
+/**
+ * @brief Set gpu device id
+ *
+ * @param gpu_id
+ *
+ * Sets gpu id using cudaSetDevice
+ *
+ * @return: cudaError_t return status
+ */
+TRTORCH_API cudaError_t set_device(const int gpu_id);
```

Comment:
We should use the device struct to set device here I think

Reply:
Also we should not return the cudaError_t, maybe we could do something else here or handle the cuda error for the user?

Reply:
This is intended for runtime. Using other Device info such as dla_core, allow_gpu_fallback won't be applicable here. These configurable parameters are required during building time only. Hence removed from here.

Device info attributes are already added in ExtraConfig during build time.

Return status would help end-users to debug errors (such as incorrect device ordinal) first hand. cudaError_t return type API is already released from CUDA

Reply:
We should still make this void or something or develop a TRTorch Error Type